src/Processors/Formats/Impl/DWARFBlockInputFormat.cpp

```diff
@@ -0,0 +1,792 @@
+            {
+                if (thread_group)
+                    CurrentThread::attachToGroupIfDetached(thread_group);
+                SCOPE_EXIT_SAFE(if (thread_group) CurrentThread::detachFromGroupIfNotDetached(););
+                try
+                {
+                    setThreadName("DWARFDecoder");
+
+                    std::unique_lock lock(mutex);
+                    while (!units_queue.empty() && !is_stopped)
+                    {
+                        if (delivery_queue.size() > num_threads)
+                        {
+                            wake_up_threads.wait(lock);
+                            continue;
+                        }
+                        UnitState unit = std::move(units_queue.front());
+                        units_queue.pop_front();
+                        ++units_in_progress;
+
+                        lock.unlock();
+
+                        size_t offset_before = unit.offset;
+                        Chunk chunk = parseEntries(unit);
+                        size_t offset_after = unit.offset;
+
+                        lock.lock();
+
+                        --units_in_progress;
+                        if (chunk)
+                        {
+                            delivery_queue.emplace_back(std::move(chunk), offset_after - offset_before);
+                            deliver_chunk.notify_one();
+                        }
+                        if (!unit.eof())
+                            units_queue.push_front(std::move(unit));
+                    }
+                }
+                catch (...)
+                {
+                    std::lock_guard lock(mutex);
+                    background_exception = std::current_exception();
+                    deliver_chunk.notify_all();
+                }
+            });
+}
+
+void DWARFBlockInputFormat::stopThreads()
+{
+    {
+        std::unique_lock lock(mutex); // required even if is_stopped is atomic
+        is_stopped = true;
+    }
+    wake_up_threads.notify_all();
+    if (pool)
+        pool->wait();
+}
+
+static inline void throwIfError(llvm::Error & e, const char * what)
+{
+    if (!e)
+        return;
+    throw Exception(ErrorCodes::CANNOT_PARSE_DWARF, "Failed to parse {}: {}", what, llvm::toString(std::move(e)));
+}
+
+Chunk DWARFBlockInputFormat::parseEntries(UnitState & unit)
+{
+    const auto & header = getPort().getHeader();
+    const auto & column_name_to_idx = getColumnNameToIdx();
+    std::array<bool, COL_COUNT> need{};
+    for (const std::string & name : header.getNames())
+        need[column_name_to_idx.at(name)] = true;
+    auto form_params = unit.dwarf_unit->getFormParams();
+
+    /// For parallel arrays, we nominate one of them to be responsible for populating the offsets vector.
+    if (need[COL_ATTR_FORM] || need[COL_ATTR_INT] || need[COL_ATTR_STR])
+        need[COL_ATTR_NAME] = true;
+    if (need[COL_ANCESTOR_OFFSETS])
+        need[COL_ANCESTOR_TAGS] = true;
```

Comment:
can these be a simple assignment?
```suggestion
    need[COL_ATTR_NAME] = need[COL_ATTR_FORM] || need[COL_ATTR_INT] || need[COL_ATTR_STR];
    need[COL_ANCESTOR_TAGS] = need[COL_ANCESTOR_OFFSETS];
```

Reply:
(That would be `|=`, which annoyingly produces compiler warnings-treated-as-errors for bool. Will change to `x = x || y`, which still looks nicer than the ifs, I guess.)